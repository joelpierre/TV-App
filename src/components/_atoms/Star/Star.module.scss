.Star {
  position: relative;
  font-size: inherit;

  &::before {
    content: '\2605';
    color: $color-tint-gamma;
  }

  &--is-active {
    &::before {
      color: gold;
    }
  }
}
