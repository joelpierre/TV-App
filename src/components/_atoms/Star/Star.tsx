import React from 'react';
import classNames from 'classnames';

import styles from './Star.module.scss';

interface IStarProps {
  className?: string;
  isActive?: boolean;
}

const Star: React.FunctionComponent<IStarProps> = ({
  className,
  isActive = false
}) => {
  return (
    <span
      className={classNames(styles.Star, className, {
        [styles['Star--is-active']]: isActive
      })}
    />
  );
};

export default Star;
