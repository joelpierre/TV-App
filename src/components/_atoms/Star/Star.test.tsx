import React from 'react';
import { render, RenderResult } from 'test-utils';

import Star from './Star';

const setup = (props = {}): RenderResult => {
  const setupProps = { ...props };
  return render(<Star {...setupProps} />);
};

describe('<Star/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
