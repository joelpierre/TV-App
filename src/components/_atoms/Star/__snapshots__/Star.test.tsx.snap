// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star/> should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="Star"
  />
</DocumentFragment>
`;
